arbiter-core/src/data_collection.rs

```diff
@@ -73,93 +85,59 @@ impl EventLogger {
-            while let Some(Ok(log)) = stream.next().await {
-                let serialized = serde_json::to_string(&log).unwrap();
-                let deserialized: BTreeMap<String, Value> =
-                    serde_json::from_str(&serialized).unwrap();
-                let (key, value) = deserialized.iter().next().unwrap();
-                let file_name = event_dir.join(format!("{}.csv", key));
-                let file_key = file_name.to_str().unwrap();
-                let file_value = files.get(file_key);
-                let toggle_written_columns = columns_written.get(file_key).unwrap_or(&false);
-                if file_value.is_none() {
-                    files.insert(
-                        file_key.into(),
-                        tokio::fs::OpenOptions::new()
-                            .write(true)
-                            .create(true)
-                            .truncate(true)
-                            .open(&file_name)
-                            .await
-                            .unwrap(),
-                    );
-                }
-
-                let file = files.get_mut(file_key).unwrap();
-
-                if toggle_written_columns == &true {
-                    let values = value
-                        .as_object()
-                        .unwrap()
-                        .values()
-                        .map(|x| x.to_string())
-                        .collect::<Vec<String>>()
-                        .join(",");
-                    file.write_all(values.as_bytes()).await.unwrap();
-                    file.write_all("\n".as_bytes()).await.unwrap();
-                } else {
-                    columns_written.entry(file_key.into()).or_insert(true);
-                    let columns = value
-                        .as_object()
-                        .unwrap()
-                        .keys()
-                        .map(|x| x.to_string())
-                        .collect::<Vec<String>>()
-                        .join(",");
-                    file.write_all(columns.as_bytes()).await.unwrap();
-                    file.write_all("\n".as_bytes()).await.unwrap();
-                    let values = value
-                        .as_object()
-                        .unwrap()
-                        .values()
-                        .map(|x| x.to_string())
-                        .collect::<Vec<String>>()
-                        .join(",");
-                    file.write_all(values.as_bytes()).await.unwrap();
-                    file.write_all("\n".as_bytes()).await.unwrap();
-                }
-                continue;
-            }
-        });
+        // Grab the connection from the client and add a new event sender so that we
+        // have a distinct channel to now receive events over
+        let event_transmuted: EventTransmuted<Arc<RevmMiddleware>, RevmMiddleware, D> =
+            unsafe { transmute(event) };
```

Comment:
idk what this does or why its here, maybe a comment?

Reply:
Same, is the kinda same unsafe stuff we did with resolving pending transactions?

Reply:
lets you access private fields

Reply:
@Alexangelj if you're talking about the transmute, it is essentially what Matt said.

So you can take an `Event<Arc<RevmMiddleware>, RevmMiddleware, D>` and cast it into the `EventTransmuted<Arc<RevmMiddleware, RevmMiddleware, D>` which then gives us direct access to the fields of the struct.